specs/interop/predeploys.md

```diff
@@ -254,6 +288,77 @@ Note that the `relayMessage` function is `payable` to enable relayers to earn in
 To enable cross chain authorization patterns, both the `_sender` and the `_source` MUST be exposed via `public`
 getters.
 
+#### Sending Back Messages Hashes
+
+When sending back a message that failed to be relayed on the destination chain
+to the source chain, it's crucial to ensure the message can only be sent back
+to the `L2ToL2CrossDomainMessenger` contract in its source chain.
+
+This function has no auth, which allows anyone to send back a given message hash.
+The `RETURN_DELAY` variable is added to give the users enough time to replay their
+failed messages and to prevent malicious actors from performing a griefing attack
+by sending messages back immediately.
+
+Once the message is returned to the source chain, the message on the local chain is set
+as successful in the `successfulMessages` mapping to ensure non-replayability.
+
+```solidity
+function revert(bytes32 _messageHash) external nonReentrant {
+    if (successfulMessages[messageHash]) revert MessageAlreadyRelayed();
+
+    (uint256 messageTimestamp, uint256 messageSource) = failedMessages[messageHash];
+
+    // Return delay is necessary to avoid griefing attacks by providing users with time to
+    // replay their transactions give this function has no auth
+    if (block.timestamp <  messageTimestamp + RETURN_DELAY) revert DelayHasNotEnsued();
```

Comment:
There is a constants table above, could we define the value of `RETURN_DELAY` in it?